src/main/java/com/minecolonies/coremod/client/gui/WindowCitizen.java

```diff
@@ -367,8 +367,8 @@ public void onOpened()
 
         createHealthBar();
         createSaturationBar();
-        createXpBar();
-        createSkillContent();
+        createXpBar(citizen, this);
+        createSkillContent(citizen, this);
 
         resourceList.setDataProvider(new ScrollingList.DataProvider()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this anonymous class number of lines from 65 to at most 40, or make it a named class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1188)